src/Sylius/Bundle/ShopBundle/Resources/config/services/twig/component.xml

```diff
@@ -155,5 +155,19 @@
                 template="@SyliusShop/product/show/page/product_listing/association.html.twig"
             />
         </service>
+
+        <service
+            id="sylius_shop.twig.component.locale_switcher"
+            class="Sylius\Bundle\ShopBundle\Twig\Component\Common\LocaleSwitcher"
+        >
+            <argument type="service" id="sylius.context.locale" />
+            <argument type="service" id="sylius.locale_provider" />
+
+            <tag
+                name="twig.component"
+                key="sylius_shop:common:locale_switcher"
+                template="@SyliusShop/shared/layout/base/header/top_bar/locale_switch.html.twig"
```

Comment:
The only questionable thing for me is either the name of the component and/or the template placement, but we can think about it at a later time when upgrading the currency switcher etc.